Grammarly suggestions

SergioGasquez opened the pull request.

f-hollow commented on src/tooling/simulating/index.md:
```diff
@@ -4,7 +4,7 @@ Simulating projects can be very handy. It allows users to test projects using CI
 At the moment, there are a few ways of simulating Rust projects on Espressif chips, all of them have some limitations, but it's quickly evolving and getting better every day.
 
-In this chapter, we will discuss the different simulation tools that are available.
+In this chapter, we will discuss the different available simulation tools.
```
I think we can omit "different", because nothing in this sentence implies that the tools are very similar.

However, it might be good to emphasize that the tools we are about to go through are available at this time. Because new tools might appear in the future.

How about:

... we will discuss currently available simulation tools.

f-hollow commented on src/writing-your-own-application/generate-project/esp-idf-template.md:
```diff
@@ -91,7 +91,7 @@ cargo run
 This builds the code according to the configuration and executes [`espflash`][espflash]  to flash the code to the board.
 
-Since our [`runner` configuration][runner-config] also passes the `--monitor` argument to [`espflash`][espflash]  we can see what the code is printing.
+Since our [`runner` configuration][runner-config] also passes the `--monitor` argument to [`espflash`][espflash] we can see what the code is printing.
```
Looks like all these in-line code snippets disoriented Grammarly. This is the complex sentence with a dependent clause followed by an independent clause. That's why you need a comma before "we can see".

... passes the `--monitor` argument to [`espflash`][espflash], we can see ...

f-hollow commented on src/writing-your-own-application/generate-project/esp-template.md:
```diff
@@ -1,7 +1,7 @@
 
 Now that we know how to [generate a no_std project][generate-no-std], let's inspect what the generated
-project contains, try to understand every part and run it.
+project contains, try to understand every part and, run it.
```
The serial comma (the Oxford comma) should be added before and. Suggest updating:

... try to understand every part of it, and run it.

f-hollow commented on src/writing-your-own-application/generate-project/esp-template.md:
```diff
@@ -39,10 +39,10 @@ Before going further, let's see what these files are for.
     - Tells `git` which folders and files to ignore
 - [Cargo.toml][cargo-toml]
-    - The usual Cargo manifest declaring some meta-data and dependencies of the project
+    - The usual Cargo manifest declares some meta-data and dependencies of the project
```
I suggest that you keep "declaring" here.

I think Grammarly treated this as a sentence and proposed an update. However, these are descriptions of files that are mostly written as sentence fragments and noun phrases.

So this edit is not necessary, but good in terms of following the same style with surrounding text.

f-hollow commented on src/writing-your-own-application/generate-project/esp-template.md:
```diff
@@ -39,10 +39,10 @@ Before going further, let's see what these files are for.
 - LICENSE-APACHE, LICENSE_MIT
     - Those are the most common licenses used in the Rust ecosystem
-    - If you want to apply a different license, you can delete these files and change the license in `Cargo.toml`
+    - If you want to apply for a different license, you can delete these files and change the license in `Cargo.toml`
```
Do you need to apply to some institution to get a license for your open source project?

If not, then suggest sticking with "apply a different license" or "use a different license".

f-hollow commented on src/writing-your-own-application/index.md:
```diff
@@ -4,7 +4,7 @@ With the appropriate Rust compiler and toolchain installed, you're now ready to
 You can write an application in the following ways:
 
-- (**Strongly recommended**) Generate from a template: Gives you a configured project, saves time and prevents possible errors.
+- (**Strongly recommended**) Generate from a template: Gives you a configured project, saves time and, prevents possible errors.
```
This is a serial comma again. It should come before "and".

Not sure why Grammarly gives this kind of suggestions...

f-hollow reviewed (approved):
Pointed out a couple nitpicks here and there.

All in all, I see that these tools spotted quite a few typos and proposed some grammar / syntax corrections.

Seems like they are worth using as proofreading does not take much time, and the content becomes more pleasant to read.

Thank you for trying the proofreading tools!

SergioGasquez commented:
Just pushed https://github.com/esp-rs/book/commit/484f4f6b4ed630239fbb018aae659961a211a247 addressing your comments to the branch, not sure why the PR is not being updated with that commit.
